### PR TITLE
Added DBMigrationsApp

### PR DESCRIPTION
**Pull request by Korgath**
## Description

Added DBMigrationsApp project to Metering point

### CODE REVIEW

**Review by MrDach — APPROVED** [review]

File: source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/Helpers/ConnectionStringFactory.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+using System.Linq;
+
+namespace Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp.Helpers
+{
+    public static class ConnectionStringFactory
+    {
+        private const string DefaultConnectionString =
+            "Server=(local); Database=MasterData; Trusted_connection=true";
+
```
Comment (kft) [inline source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/Helpers/ConnectionStringFactory.cs:23]: Is a default connection needed ? Should it not be something that is to be configured ?

File: source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/Scripts/Model/202105051000 Create OutBoxMessages Table.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    [Data] [nvarchar](max) NOT NULL,
+    [Category] [nvarchar](50) NOT NULL,
+    [CreationDate] [datetime2](7) NOT NULL,
+    [ProcessedDate] [datetime2](7) NULL,
+    CONSTRAINT [PK_OutboxMessages] PRIMARY KEY NONCLUSTERED
+(
+[Id] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
+    ) ON [PRIMARY] TEXTIMAGE_ON [PRIMARY]
+    GO
```
Comment (kft) [inline source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/Scripts/Model/202105051000 Create OutBoxMessages Table.sql:14]: @thbaan: Could you please have a look at this ?

**Review by kft — DISMISSED** [review]
Just a couple of comments.

File: source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/Helpers/ConnectionStringFactory.cs (hunk 1 above)
Comment (Korgath) [reply source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/Helpers/ConnectionStringFactory.cs]: You are right. The string come in as an arg. There are even error handling done if there is no connectionstring later in the process. Removed the defualt one

File: source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/Scripts/Model/202105051000 Create OutBoxMessages Table.sql (hunk 2 above)
Comment (Korgath) [reply source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/Scripts/Model/202105051000 Create OutBoxMessages Table.sql]: Unique cluster index added on recordID

**Review by MrDach — APPROVED** [review]

**Review by kft — APPROVED** [review]

File: source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/Helpers/UpgradeFactory.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+using System;
+using System.Reflection;
+using DbUp;
+using DbUp.Engine;
+
+namespace Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp.Helpers
+{
+    public static class UpgradeFactory
+    {
+        // TODO: Delete this method once scripts have been ported to new format.
```
Comment (cmgrd) [inline source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/Helpers/UpgradeFactory.cs:24]: Isn't this the first script?

**Review by cmgrd — CHANGES_REQUESTED** [review]

**Review by thbaan — APPROVED** [review]

File: source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# ApplyDBMigrationsApp
+
+## Purpose
+
+During the cause of time, the structure contained in a SQL server will change, be it tables, views, stored procedures etc.
```
Comment (erduna) [inline source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/README.md:5]:
```suggestion
During the course of time, the structure contained in a SQL server will change, be it tables, views, stored procedures etc.
```

File: source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+The purpose of the migrations app is to make sure that the database structures are up to date.
+
+As such, it contains a number of scripts needed to go from an empty database to a database containing all the needed structure for the newest version. That is, the entire update history.
+
+After the SQL server has been deployed this tool can be run and will then ensure that all the scripts available that has not already been run on the server will be executed.
```
Comment (erduna) [inline source/Energinet.DataHub.MeteringPoints.ApplyDBMigrationsApp/README.md:11]:
```suggestion
After the SQL server has been deployed this tool can be run and will then ensure that all the scripts available that have not already been run on the server will be executed.
```

**Review by cmgrd — APPROVED** [review]

**Review by MrDach — APPROVED** [review]

**Pull request merged**